FrontEnd/pages/index.vue

```diff
@@ -1,49 +1,55 @@
 <template>
   <div>
-    <MainSection title="Home" :loading="loading">
+    <MainSection title="Home">
       <div class="border-b" :class="borderColorConfig">
         <TweetForm />
       </div>
-      <!-- <TweetListFeild page="Home" :tweets="tweets"/> -->
-      <TweetListFeild page="Home"/>
+      <TweetListFeild page="Home" :tweets="tweets" :loading="loading" />
     </MainSection>
   </div>
 </template>
 
 <script setup>
-import { useUserStore } from "../stores/useUserStore";
-// import { usePostStore } from "~/stores/usePostStore";
-
-
-const {borderColorConfig} = useTailwindConfig()
-// const { getTweets } = usePost();
-
-
-const user = useUserStore();
-const loading = ref(false);
-// const postDetails = usePostStore();
-
-
-onMounted(async() => {
-  user.initialize();
-  // try {
-  //   loading.value = true;
-  //   const response = await getTweets();
-  // } catch (error) {
-  //   console.log(error);
-  // } finally {
-  //   loading.value = false;
-  // }
+import { usePostStore } from "~/stores/post";
+import { useCookie } from '#app';
+
+
+const { borderColorConfig } = useTailwindConfig();
+const { getTweets } = usePost();
+
+const postDetails = usePostStore();
+const tweets = ref([]);
+const loading = ref(true);
+
+const {
+  data: tweetArray,
+  error,
+  status,
+} = await useAsyncData("tweets", async () => {
+  try {
+    return await getTweets();
+  } catch (err) {
+    console.error("Error fetching tweets:", err);
+    throw err;
+  }
 });
 
-// const tweets = postDetails.posts
-
+if (error.value) {
+  console.error("Failed to load tweets:", error.value.message);
+} else {
+  postDetails.setPosts(tweetArray.value.data);
+  tweets.value = postDetails.posts;
+}
+
+loading.value = status.value !== "success";
+watch(
```

Comment:
avoid watchers

Reply:
used computed property for reactivity while adding new tweets instead of watchers  for desired output.